### PR TITLE
feat(service): add new service resource

### PR DESCRIPTION
**Pull request by saviogl**
Adding new `Service` resource as defined [here](https://docs.opsgenie.com/docs/service-api)

I created a trial account on `OpsGenie` to see if I'd be able to run the integration tests but I kept running into some permission issues. I guess my free trial account wasn't enough.

I'm hoping to get some help to validate the integration tests and the new resources. 

#61

### CODE REVIEW

**Review by ffahri — CHANGES_REQUESTED** [review]
It seems our go-SDK-v2 is missing some fields. I created a PR for SDK. 
And we don't need visibility its deprecated and documents will be changed today.

Here is the current status of get Result
```
{
  "data": {
    "id": "**",
    "name": "Payment",
    "description": "",
    "teamId": "**",
    "tags": [],
    "links": {
      "web": "https://deneme123.app.opsgenie.com/service/**/status",
      "api": "https://api.opsgenie.com/v1/services/**"
    },
    "isExternal": false
  },
  "took": 0.119,
  "requestId": "**"
}
```

**Comment by ffahri** [comment]
https://github.com/opsgenie/opsgenie-go-sdk-v2/pull/47
related pr

File: website/docs/r/service.html.markdown [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+```hcl
+resource "opsgenie_team" "payment" {
+  name        = "example"
+  description = "This team deals with all the things"
+}
+
+resource "opsgenie_service" "this" {
+  name  = "Payment"
+  team_id = "$opsgenie_team.this.id"
```
[the remark this answers is not in the document]
Comment (saviogl) [reply website/docs/r/service.html.markdown]: @ffahri Good catch on the doc

File: go.sum [hunk 2]
```diff
@@ -178,6 +178,7 @@ github.com/opsgenie/opsgenie-go-sdk-v2 v1.0.6 h1:jbPcx0GlwyDkFqFPVQQ5SXCHMVmftLo
 github.com/opsgenie/opsgenie-go-sdk-v2 v1.0.6/go.mod h1:f0ezb0R/mrB9Hpm5RrIS6EX3ydjsR2nAB88nYYXZcNY=
 github.com/opsgenie/opsgenie-go-sdk-v2 v1.0.7-0.20200406122459-ce3f70f22ab7 h1:+cTGQp1QRUWH+YGgh2b4sO/Y5g86PTpVEIpVFnEor6E=
 github.com/opsgenie/opsgenie-go-sdk-v2 v1.0.7-0.20200406122459-ce3f70f22ab7/go.mod h1:VOkJ7STzYj+nXRhMcBTcmt8uZZ17KZKJdZtJpgHLbT8=
+github.com/opsgenie/opsgenie-go-sdk-v2 v1.1.1 h1:0y6RTB5SNcf0lhdVyJFPU/bTagUlToKpW4AGWMB7ZBw=
```
Comment (ffahri) [inline go.sum:181]: could you change this to master 
my pr has been merged into go-sdk-v2 and get method will work on the master branch

**Review by ffahri — CHANGES_REQUESTED** [review]

File: go.sum (hunk 2 above)
Comment (ffahri) [reply go.sum:181]: wrong file my bad, in the go.mod file*

Comment (saviogl) [reply go.sum:181]: @ffahri Just did but I'm getting some issues with `notification` resource:

image

Comment (saviogl) [reply go.sum:181]:
```bash
❯ make test
==> Checking that code complies with gofmt requirements...
go test -i $(go list ./... |grep -v 'vendor') || exit 1
# github.com/terraform-providers/terraform-provider-opsgenie/opsgenie
opsgenie/resource_opsgenie_notification_policy.go:520:22: cannot use config["until_minute"].(int) (type int) as type *int in assignment
opsgenie/resource_opsgenie_notification_policy.go:521:20: cannot use config["until_hour"].(int) (type int) as type *int in assignment
make: *** [GNUmakefile:12: test] Error 1

```

Comment (ffahri) [reply go.sum:181]:
```
		action.UntilMinute = config["until_minute"].(*int)
		action.UntilHour = config["until_hour"].(*int)
```
should do the trick

Comment (saviogl) [reply go.sum:181]: @ffahri Cool, dep updated and pushed to the PR. As for the `Service API` fields I noticed Sat that the `sdk` didn't support the `tags`  field that's why I left it out.

Comment (ffahri) [reply go.sum:181]: last one 😄 
```
cannot use config["until_minute"].(int) (type *int) as type int in assignment
```

you forgot to push policy.go on the vendor

Comment (saviogl) [reply go.sum:181]: @ffahri Is this what you mean https://github.com/terraform-providers/terraform-provider-opsgenie/pull/115/commits/b7cd2829f87a8f9bf79cd0af22bcbb70d8a62ca6

Comment (ffahri) [reply go.sum:181]: yes, but you can remove it since its nothing related to this pr, we can add this after this pr

Comment (ffahri) [reply go.sum:181]: completeTest contains 
```
	visibility = "%s"

```
after we removed that e2e will pass and we can merge

Comment (saviogl) [reply go.sum:181]: @ffahri Cool. As you suggested, removed the `b7cd282` commit regarding vendoring. 

Thanks for catching the `visibility` field on the e2e tests as well

**Comment by saviogl** [comment]
@ffahri It seems that Travis build is failing because I excluded this commit `b7cd282` should I bring it back or are we good?

**Comment by ffahri** [comment]
Thank you for your contribution @saviogl

**Pull request merged**